tests/drivers/clock_control/nrf_clock_calibration/src/test_nrf_clock_calibration.c

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) 2019, Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <ztest.h>
+#include <clock_control.h>
+#include <nrf_clock.h>
+#include <logging/log.h>
+LOG_MODULE_REGISTER(test);
+
+#ifndef CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC
+#error "LFCLK must use RC source"
+#endif
+
+#if CONFIG_CLOCK_CONTROL_NRF_CALIBRATION_PERIOD != 1
+#error "Expected 250ms calibration period"
+#endif
+
+static void turn_off_clock(struct device *dev)
+{
+	int err;
+
+	do {
+		err = clock_control_off(dev, 0);
+	} while (err == 0);
+}
+
+/* Test checks if calibration clock is running and generates interrupt as
+ * expected  and starts calibration. Validates that HF clock is turned on
+ * for calibration and turned off once calibration is done.
+ */
+static void test_clock_calibration(void)
+{
+	struct device *hfclk_dev =
+		device_get_binding(DT_INST_0_NORDIC_NRF_CLOCK_LABEL  "_16M");
+	struct device *lfclk_dev =
+		device_get_binding(DT_INST_0_NORDIC_NRF_CLOCK_LABEL  "_32K");
+	int key;
+	u32_t cnt = 0;
+	u32_t max_cnt = 1000;
+
+	turn_off_clock(hfclk_dev);
+	turn_off_clock(lfclk_dev);
+
+	clock_control_on(lfclk_dev, NULL);
+
+	/* It may be required to restart CT. Allow that by 1ms without lock */
+	k_busy_wait(1000);
+
+	key = irq_lock();
+	while (nrf_clock_event_check(NRF_CLOCK_EVENT_CTTO) == 0) {
+		k_busy_wait(1000);
+		cnt++;
+		if (cnt == max_cnt) {
+			irq_unlock(key);
+			clock_control_off(lfclk_dev, NULL);
+			zassert_true(false, "");
+		}
+	}
+
+	zassert_within(cnt, 250, 10, "Expected 250ms period");
```

Comment:
Am I correct assuming that the first calibration will start 250ms after clock startup? If so, shouldn't we start calibration as early as possible in order to provide stable clock?

Reply:
i've changed the behavior. first calibration is performed immediately after starting LF clock.